diff --git a/lib/Cloudflare/Workers/WorkerProvider.js b/lib/Cloudflare/Workers/WorkerProvider.js
index 622b336..9afeab1 100644
--- a/lib/Cloudflare/Workers/WorkerProvider.js
+++ b/lib/Cloudflare/Workers/WorkerProvider.js
@@ -3021,7 +3021,7 @@ export const LiveWorkerProvider = () => Provider.effect(Worker, Effect.gen(funct
             compatibilityFlags: compatibility.flags,
             containers: metadataContainers.length > 0 ? metadataContainers : undefined,
             keepAssets,
-            keepBindings: undefined,
+            keepBindings: ["secret_text", "secret_key"],
             limits: news.limits,
             logpush: news.logpush,
             mainModule: bundle.main,
diff --git a/src/Cloudflare/Workers/WorkerProvider.ts b/src/Cloudflare/Workers/WorkerProvider.ts
index 798e849..fb6f879 100644
--- a/src/Cloudflare/Workers/WorkerProvider.ts
+++ b/src/Cloudflare/Workers/WorkerProvider.ts
@@ -4195,7 +4195,7 @@ export const LiveWorkerProvider = () =>
           containers:
             metadataContainers.length > 0 ? metadataContainers : undefined,
           keepAssets,
-          keepBindings: undefined,
+          keepBindings: ["secret_text", "secret_key"],
           limits: news.limits,
           logpush: news.logpush,
           mainModule: bundle.main,
